Move katello and foreman to roles

tstrachota opened the pull request:
- new roles foreman and katello, original playbooks of the same name just require the new roles
- katello_provisioning split into foreman and katello part so that they can be used independently

ehelms commented:
The meta role look fine here. The bulk of this is the change to the provisioning setup so I'll let @stbenjam review that.

xprazak2 commented:
katello_provisioning playbook is meant for hosts that have already katello set up, right? As the number of playbooks grows, maybe we could add a short comment to what a playbook actually does, as it may not be obvious just from the list of roles. I think it would help the newcomers to quickly define their boxes.

xprazak2 commented on playbooks/roles/libvirt_setup/tasks/main.yml:
```diff
@@ -22,6 +22,10 @@
+  yum:
+    name: libvirt-python
+
 - name: 'create the provisioning libvirt network'
```
When running outside of vagrant box, I got error on Centos 7.2:
"The lxml module is not importable. Check the requirements."

The following helped:
```yaml
- name : 'install python-lxml'
  yum:
    name: python-lxml
```

xprazak2 commented on playbooks/roles/foreman_provisioning/tasks/main.yml:
```diff
@@ -0,0 +1,215 @@
+# Subnet
+- name: 'find subnet'
+  shell: >
+    {{ foreman_provisioning_hammer }} subnet info --name "{{ foreman_provisioning_network }}/24"
```
foreman_provisioning_network is not defined

xprazak2 commented on playbooks/roles/katello_provisioning/tasks/main.yml:
```diff
@@ -1,18 +1,184 @@
+- name: 'set enable-puppet4 parameter'
+  shell: >
+      {{ katello_provisioning_hammer }} hostgroup set-parameter
+      --hostgroup "Forklift CentOS 7"
```
should be "Forklift Katello CentOS 7" or "Forklift CentOS 7 Mirror"

xprazak2 commented:
I noticed that all hostgroups have no value for PXE loader, is that intentional?

tstrachota commented:
I rebased and fixed the comments.
I'll add a role for setting hammer credentials as mentioned in https://github.com/theforeman/forklift/pull/397#discussion_r105921809

tstrachota commented:
Updated, I added role `hammer_connection` that adds custom configuration file with credentials into `~/.hammer/cli.modules.d/`.

ehelms commented:
Feels a bit like `hammer_connection` could just become `hammer_config` or more generically `hammer` ?

tstrachota commented:
@ehelms I was struggling with finding the right name. I didn't indend the role to be universal tool for configuring anything in hammer, so I excluded `hammer_config`. `hammer` seemed too generic.

Maybe `hammer_credentials`? I absolutely ok with renaming it to anything that majority of devs like.
So, other people, opinions?

xprazak2 commented:
I do not have any objections against `hammer_connection`,  `hammer_credentials` seems a little better but I do not have any objections either way. @tstrachota, could you rebase pls?

tstrachota commented:
@xprazak2 rebased. I renamed `hammer_connection` to `hammer_credentials`. Please review.

xprazak2 commented on playbooks/roles/hammer_connection/tasks/main.yml:
```diff
@@ -12,4 +12,4 @@
 - name: 'Deploy configuration file with connection setup'
   template:
     src: zzz_foreman_forklift_connection.yml.j2
```
This needs to be `zzz_foreman_forklift_credentials.yml.j2`

xprazak2 commented:
I think you need to rename the folder to hammer_credentials as well.

ekohl commented on playbooks/roles/foreman_installer/tasks/module_pr.yml:
```diff
@@ -12,6 +12,22 @@
+  tags:
+    - packages
+
+- name: 'install yard gem'
```
This is no longer needed, the above code coverages this. It has been refactored.

ekohl commented on playbooks/roles/foreman_installer/tasks/packages.yml:
```diff
@@ -1,11 +1,30 @@
 ---
+- name: 'Update packages'
```
This has moved to the ```update_os_packages``` role.

ekohl commented on playbooks/roles/foreman_installer/tasks/packages.yml:
```diff
@@ -1,11 +1,30 @@
+  tags:
+    - packages
+
+- name: 'Install foreman-release-scl'
```
Has moved to the ```foreman_repositories``` role.

ekohl commented:
Looks like my issues were fixed in a later commit, but it still needs a rebase.

tstrachota commented:
@ekohl rebased
@xprazak2 opps, I'm sorry for the mistake with rename, it's fixed now

xprazak2 commented on playbooks/roles/foreman_provisioning/tasks/main.yml:
```diff
@@ -0,0 +1,216 @@
+  set_fact:
+    kickstart_templates: "{{ kickstart_templates_json.stdout|from_json }}"
+
+- name: 'associate kickstart templates to CentOS 7'
```
nit: templates are associated with CentOS 7, but they are not assigned to os

xprazak2 commented on playbooks/roles/katello_provisioning/tasks/main.yml:
```diff
@@ -1,18 +1,184 @@
+    --query-organization "{{ katello_provisioning_organization }}"
+    --content-source-id {{ foreman_provisioning_smart_proxy.Id }}
+    --kickstart-repository-id 1
+    --operatingsystem "CentOS_Linux 7"
```
`CentOS_Linux 7` has no partition table

tstrachota commented:
@xprazak2 I pushed one more commit that should fix the template association.

ehelms commented:
@ekohl @xprazak2 I believe last reviews and/or approvals are on y'all.

xprazak2 commented on playbooks/roles/katello_provisioning/tasks/main.yml:
```diff
@@ -1,18 +1,236 @@
+    --content-source-id {{ foreman_provisioning_smart_proxy.Id }}
+    --kickstart-repository-id 1
+    --operatingsystem "CentOS_Linux 7"
+    --medium 'CentOS mirror'
```
Is this hostgroup supposed to use synced content? If so, this should be removed.

tstrachota commented:
@xprazak2 I fixed the issue with kickstart repo id and rebased, please take a look.

ehelms commented:
I merged a re-org just now that will cause you to have to rebase and move your changes to the top level roles directory.

https://github.com/theforeman/forklift/pull/476

xprazak2 commented:
Everything seems to be working now, so all we need is a rebase and we are ready to merge.

ekohl commented on playbooks/roles/foreman_provisioning/tasks/main.yml:
```diff
@@ -0,0 +1,226 @@
+    --foreman-proxy-tftp-managed false
+    {{ foreman_provisioning_installer_options | join(" ") }}
+
+- name: 'restart foreman-proxy'
```
Shouldn't the installer take care of this?

tstrachota replied:
Honestly, I don't know. I just moved the code from here https://github.com/theforeman/forklift/pull/399/files#diff-1574be18ceccad8d4354975e4cf6be59L29 to another location. I suppose the author of the task had a reason to add it. If we want to do a complete audit of all tasks in the roles, I'd treat it as a separate issue.

ekohl commented on playbooks/roles/foreman_provisioning/tasks/main.yml:
```diff
@@ -0,0 +1,226 @@
+- name: 'restart foreman-proxy'
+  service: name=foreman-proxy state=restarted
+
+- name: 'restart foreman'
```
Shouldn't the installer take care of this?

tstrachota replied:
ditto

ekohl commented on playbooks/roles/foreman_provisioning/tasks/main.yml:
```diff
@@ -0,0 +1,226 @@
+- name: 'restart foreman'
+  service: name=httpd state=restarted
+
+- name: 'refresh features'
```
Shouldn't the installer take care of this?

tstrachota replied:
ditto

ekohl commented on playbooks/roles/katello/meta/main.yml:
```diff
@@ -0,0 +1,21 @@
+      - "--foreman-admin-password {{ foreman_installer_admin_password }}"
+    foreman_installer_additional_packages:
+      - katello
+
```
Can you remove these empty lines?

tstrachota replied:
removed

tstrachota commented:
@ehelms @xprazak2  I rebased and moved the roles to the top level folder.

xprazak2 commented:
👏

ekohl commented on roles/hammer_credentials/defaults/main.yml:
```diff
@@ -0,0 +1,4 @@
+---
+hammer_creadentials_username: admin
```
s/creadentials/credentials/

ekohl commented on roles/foreman_provisioning/meta/default.yml:
```diff
@@ -0,0 +1,3 @@
+---
+dependencies:
+  - role: hammer_creadentials
```
s/hammer_creadentials/hammer_credentials/

This is also a new dependency and I'd prefer to use ```foreman::cli``` through the installer since we don't have to replicate things then.

ekohl commented on roles/hammer_credentials/defaults/main.yml:
```diff
@@ -0,0 +1,4 @@
+---
+hammer_creadentials_username: admin
+hammer_creadentials_password: "{{ foreman_installer_admin_password | default('changeme') }}"
+hammer_creadentials_host: https://localhost
```
Can we use a fact like ```ansible_fqdn```?

ekohl commented on roles/hammer_credentials/templates/zzz_foreman_forklift_credentials.yml.j2:
```diff
@@ -0,0 +1,8 @@
+  :password: {{ hammer_creadentials_password }}
+
+:ssl:
+  :verify_ssl: false
```
I'd prefer to enable SSL verification by default. Production should also use it and this could help us detect problems early on. If needed you can point to a CA file, which is what ```foreman::cli``` in the ```puppet-foreman``` module does.

tstrachota commented:
@ekohl thanks, updated.
